src/lib/core/CASEAuthTag.h

```diff
@@ -98,13 +158,12 @@ struct CATValues
     bool CheckSubjectAgainstCATs(NodeId subject) const
     {
         VerifyOrReturnError(IsCASEAuthTag(subject), false);
+        CASEAuthTag candidate = CASEAuthTagFromNodeId(subject);
 
         for (auto cat : values)
         {
-            // All valid CAT values are always in the beginning of the array followed by kUndefinedCAT values.
-            ReturnErrorCodeIf(cat == kUndefinedCAT, false);
-            if (((cat & kTagIdentifierMask) == (subject & kTagIdentifierMask)) &&
-                ((cat & kTagVersionMask) >= (subject & kTagVersionMask)))
+            if ((cat != kUndefinedCAT) && (GetCASEAuthTagIdentifier(cat) == GetCASEAuthTagIdentifier(candidate)) &&
+                (GetCASEAuthTagVersion(candidate) > 0) && (GetCASEAuthTagVersion(cat) >= GetCASEAuthTagVersion(candidate)))
```

Comment:
Pre-existing, but this looks wrong to me.  Shouldn't this be checking that the _candidate_ version is at least the version of `cat`?  Certainly the spec has:
```
     (get_cat_id(acl_subject) == get_cat_id(isd_subject)) and
     (get_cat_version(isd_subject) >= get_cat_version(acl_subject))
```
to be considered a match.

Reply:
Per Slack discussion, this code is correct.  "subject" here is an _acl_ subject, not an ISD subject, and "cat" is one of the ISD subjects.